plugins/event_source/cmd.py

```diff
@@ -0,0 +1,92 @@
+"""
+cmd.py
+
+An ansible-rulebook event source plugin for running a custom command.
+It can send events to the rule engine from the stdout.
+
+Arguments:
+    command: str
+        The string with the command to execute
+    send_output: bool, default: false
+        Send events from the stdout.
+        Each event must be a serialized JSON string
+        ended with a newline character ('\n')
+
+Examples:
+---
+- name: Example with output proccessing
+  hosts: all
+  sources:
+    - ansible.eda.cmd:
+        command: node my_custom_plugin.js
+        send_output: true
+  rules:
+    - name: r1
+      condition: event.cmd.status == "Running"
+      action:
+        print_event:
+
+
+---
+- name: Example combined with webhook plugin
+  hosts: all
+  sources:
+    - ansible.eda.webhook:
+        port: 6080
+    - ansible.eda.cmd:
+        command: node my_custom_plugin.js --url http://localhost:6080/eda
+  rules:
+    - name: r1
+      condition: event.payload.status == "Running"
+      action:
+        print_event:
+"""
+
+import asyncio
+import json
+import logging
+from typing import Any, Dict
+
+LOGGER = logging.getLogger("ansible.eda.cmd")
+
+
+async def process_output(
+    proc: asyncio.subprocess.Process, queue: asyncio.Queue
+):
+    # proc.stdout can be None, this is not our case
+    output = await proc.stdout.readline()
+    if output:
+        try:
+            event = json.loads(output.decode())
+            await queue.put({"cmd": event})
+        except json.JSONDecodeError as e:
+            LOGGER.error("Can not deserialize JSON data: %s", e)
+
+
+async def main(queue: asyncio.Queue, args: Dict[str, Any]):
+    command = str(args["command"])
+    send_output = False
+
+    if "send_output" in args:
+        send_output = args["send_output"]
+
+    proc = await asyncio.subprocess.create_subprocess_shell(
+        command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+    )
+
+    LOGGER.info("Process started")
+
+    while proc.returncode is None:
```

Comment:
You use `PIPE` for both `stdout` and `stderr`  streams, but reading only `stdout` while process is running. This may lead to a deadlock.

Reply:
Good point. Changed to `communicate()` that according to the documentation prevents the deadlock.

Reply:
It doesn't fix the deadlock issue, because the deadlock appears earlier.

Here is the correct example of handling process `stderr` \ `stdout` stream:

```
    if send_output:
        stdout_mode = asyncio.subprocess.PIPE
    else:
        stdout_mode = asyncio.subprocess.DEVNULL

    proc = await asyncio.subprocess.create_subprocess_shell(
        command, stdout=stdout_mode, stderr=asyncio.subprocess.PIPE
    )
    
    LOGGER.info("Process started")

    if send_output:
        _, stderr = await asyncio.gather(
            process_output(proc, queue),
            proc.stderr.read()
        )
        await proc.wait()
    else:
        _, stderr = await proc.communicate()

    if proc.returncode != 0:
        LOGGER.error(
            "Command failed with code %s: %s", proc.returncode, stderr.decode()
        )

```

Reply:
I see your point. You are right, I have to deal with stderr always. Your proposal seems more reliable. I will change it and test it later.  Anyway I think I want to bring the stderr in all cases. 

Thank you very much for your suggestions.